ui/ui-components/pages/model/[modelId]/source/[sourceId]/edit.tsx

```diff
@@ -16,32 +18,79 @@ import { ErrorHandler } from "../../../../../utils/errorHandler";
 import { SuccessHandler } from "../../../../../utils/successHandler";
 import { SourceHandler } from "../../../../../utils/sourceHandler";
 import ModelBadge from "../../../../../components/badges/ModelBadge";
-import { NextPageContext } from "next";
 import { ensureMatchingModel } from "../../../../../utils/ensureMatchingModel";
+import FormMappingSelector from "../../../../../components/source/FormMappingSelector";
+import { createSchedule } from "../../../../../components/schedule/Add";
+
+interface FormData {
+  bootstrap?: {
+    propertyKey: string;
+    propertyType: string;
+  };
+  mapping?: {
+    sourceColumn: string;
+    propertyKey: string;
+  };
+  source: Pick<Models.SourceType, "name" | "options">;
+}
 
-export default function Page(props) {
-  const {
-    errorHandler,
-    successHandler,
-    sourceHandler,
-    environmentVariableOptions,
-  }: {
-    errorHandler: ErrorHandler;
-    successHandler: SuccessHandler;
-    sourceHandler: SourceHandler;
-    environmentVariableOptions: Actions.AppOptions["environmentVariableOptions"];
-  } = props;
+interface Props {
+  environmentVariableOptions: Actions.AppOptions["environmentVariableOptions"];
+  properties: Models.PropertyType[];
+  propertyExamples: Record<string, string[]>;
+  scheduleCount: number;
+  source: Models.SourceType;
+  types: string[];
+}
+
+interface InjectedProps extends NextPageContext {
+  errorHandler: ErrorHandler;
+  successHandler: SuccessHandler;
+  sourceHandler: SourceHandler;
+}
```

Comment:
This is a temp change to make the props on this page work. I created a chore in our board to make something like this available across all page components, but I did not want to apply it in this PR.